kotlin/change-basemaps/src/main/java/com/esri/arcgisruntime/sample/switchbasemaps/MainActivity.kt

```diff
@@ -0,0 +1,167 @@
+import android.support.v7.app.AppCompatActivity
+import android.view.MenuItem
+import android.view.View
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+  private var mNavigationDrawerItemTitles: Array<String>? = null
+
+  private var mDrawerToggle: ActionBarDrawerToggle? = null
+  private var mActivityTitle: String? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // inflate navigation drawer
+    mNavigationDrawerItemTitles = resources.getStringArray(R.array.vector_tiled_types)
+    // get app title
+    mActivityTitle = title.toString()
+
+    addDrawerItems()
+    setupDrawer()
+
+    supportActionBar?.apply {
+      setDisplayHomeAsUpEnabled(true)
+      setHomeButtonEnabled(true)
+      // set opening basemap title to Topographic
+      title = mNavigationDrawerItemTitles!![2]
+    }
+
+    // create a map with Topographic Basemap and set it to the map
+    mapView.map = ArcGISMap(Basemap.Type.TOPOGRAPHIC, 47.6047381, -122.3334255, 12)
+  }
+
+  /**
+   * Add navigation drawer items
+   */
+  private fun addDrawerItems() {
+    ArrayAdapter(this, android.R.layout.simple_list_item_1, mNavigationDrawerItemTitles!!).apply {
+      drawerList.adapter = this
+      drawerList.onItemClickListener =
+          AdapterView.OnItemClickListener { adapterView, view, position, id -> selectBasemap(position) }
+    }
+  }
+
+  /**
+   * Set up the navigation drawer
+   */
+  private fun setupDrawer() {
+
+    mDrawerToggle = object : ActionBarDrawerToggle(this, drawerLayout, R.string.drawer_open, R.string.drawer_close) {
+
+      /** Called when a drawer has settled in a completely open state.  */
+      override fun onDrawerOpened(drawerView: View) {
+        super.onDrawerOpened(drawerView)
+        supportActionBar!!.title = mActivityTitle
+        invalidateOptionsMenu() // creates call to onPrepareOptionsMenu()
+      }
+
+      /** Called when a drawer has settled in a completely closed state.  */
+      override fun onDrawerClosed(view: View) {
+        super.onDrawerClosed(view)
+        invalidateOptionsMenu() // creates call to onPrepareOptionsMenu()
+      }
+    }
+
+    mDrawerToggle!!.isDrawerIndicatorEnabled = true
+    drawerLayout.addDrawerListener(mDrawerToggle!!)
+  }
+
+  /**
+   * Select the Basemap item based on position in the navigation drawer
+   *
+   * @param position order int in navigation drawer
+   */
+  private fun selectBasemap(position: Int) {
+    // update selected item and title, then close the drawer
+    drawerList.setItemChecked(position, true)
+    drawerLayout.closeDrawer(drawerList)
+
+    // select basemap and title by position
+    when (position) {
```

Comment:
A few improvements could be made here (inline comments are for your understanding):
```kotlin
    val basemapTitle = mNavigationDrawerItemTitles[position]
   // moved out of when block
    supportActionBar?.title = basemapTitle
    val basemapType = Basemap.Type.valueOf(basemapTitle)
   // when on enum type (if you populate them based on my previous suggestion) is stronger
    mapView.map.basemap = when (basemapType) {
      Basemap.Type.DARK_GRAY_CANVAS_VECTOR -> Basemap.createDarkGrayCanvasVector()
      Basemap.Type.IMAGERY -> Basemap.createImagery()
      Basemap.Type.IMAGERY_WITH_LABELS -> Basemap.createImageryWithLabels()
      Basemap.Type.IMAGERY_WITH_LABELS_VECTOR -> Basemap.createImageryWithLabelsVector()
      Basemap.Type.LIGHT_GRAY_CANVAS -> Basemap.createLightGrayCanvas()
      Basemap.Type.LIGHT_GRAY_CANVAS_VECTOR -> Basemap.createDarkGrayCanvasVector()
      Basemap.Type.NATIONAL_GEOGRAPHIC -> Basemap.createNationalGeographic()
      Basemap.Type.NAVIGATION_VECTOR -> Basemap.createNavigationVector()
      Basemap.Type.OCEANS -> Basemap.createOceans()
      Basemap.Type.OPEN_STREET_MAP -> Basemap.createOceans()
      Basemap.Type.STREETS -> Basemap.createStreets()
      Basemap.Type.STREETS_NIGHT_VECTOR -> Basemap.createStreetsNightVector()
      Basemap.Type.STREETS_WITH_RELIEF_VECTOR -> Basemap.createStreetsWithReliefVector()
      Basemap.Type.STREETS_VECTOR -> Basemap.createStreetsVector()
      Basemap.Type.TOPOGRAPHIC -> Basemap.createTopographic()
      Basemap.Type.TERRAIN_WITH_LABELS -> Basemap.createTerrainWithLabels()
      Basemap.Type.TERRAIN_WITH_LABELS_VECTOR -> Basemap.createTerrainWithLabelsVector()
      Basemap.Type.TOPOGRAPHIC_VECTOR -> Basemap.createTopographicVector()
    }
```

Unfortunately, we don't have a `Basemap` constructor that takes a `Basemap.Type`. If we did, this would be trivial :/